tests/Functional/ViewHelpers/JoinViewHelperTest .php

```diff
@@ -0,0 +1,145 @@
+    {
+        yield 'template with named layout' => [
+            'arguments' => [
+                'value' => [],
+            ],
+            'src' => '<f:join value="{value}" />',
+            'expectation' => '',
+        ];
+        yield 'empty value inline' => [
+            'arguments' => [
+                'value' => [],
+            ],
+            'src' => '{value -> f:join()}',
+            'expectation' => '',
+        ];
+        yield 'empty attribute value, with separator' => [
+            'arguments' => [
+                'value' => [],
+                'separator' => ',',
+            ],
+            'src' => '<f:join value="{value}" separator="{separator}" />',
+            'expectation' => '',
+        ];
+        yield 'empty attribute value, with separatorLast' => [
+            'arguments' => [
+                'value' => [],
+                'separatorLast' => ' and ',
+            ],
+            'src' => '<f:join value="{value}" separatorLast="{separatorLast}" />',
+            'expectation' => '',
+        ];
+        yield 'value attribute' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+            ],
+            'src' => '<f:join value="{value}" />',
+            'expectation' => '123',
+        ];
+        yield 'value attribute inline array' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+            ],
+            'src' => '<f:join value="{0: \'1\', 1: \'2\', 2: \'3\'}" />',
+            'expectation' => '123',
+        ];
+        yield 'value inline' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+            ],
+            'src' => '{value -> f:join()}',
+            'expectation' => '123',
+        ];
+        yield 'value and separator set' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+                'separator' => ', ',
+            ],
+            'src' => '<f:join value="{value}" separator="{separator}" />',
+            'expectation' => '1, 2, 3',
+        ];
+        yield 'value and separatorLast set' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+                'separatorLast' => ' and ',
+            ],
+            'src' => '<f:join value="{value}" separatorLast="{separatorLast}" />',
+            'expectation' => '12 and 3',
+        ];
+        yield 'value, separator and separatorLast set' => [
+            'arguments' => [
+                'value' => [1, 2, 3],
+                'separator' => ', ',
+                'separatorLast' => ' and ',
+            ],
+            'src' => '<f:join value="{value}" separator="{separator}" separatorLast="{separatorLast}" />',
+            'expectation' => '1, 2 and 3',
+        ];
+    }
+
+    #[DataProvider('renderValidDataProvider')]
+    #[Test]
+    public function renderValid(array $arguments, string $src, string $expectation): void
+    {
+
+        $view = new TemplateView();
```

Comment:
Can you execute the rendering and testing twice here, similar to other tests? The first execution tests uncached, the second cached templates.

Reply:
added, found tests were not executed at all due to a space in the filename 🙈 - copy fail from Gerrit patch